doc/ztp/ztp_hld.md

```diff
@@ -0,0 +1,166 @@
+# Zero Touch Provisioning (ZTP)
+# High Level Design
+
+### Rev 1.0
+
+# Table of Contents
+  * [Revision](#revision)
+  * [Requirements](#requirements)
+  * [DHCP Options](#dhcp-options)
+  * [Flow](#flow)
+  * [Error Cases](#error-cases)
+  * [ZTP CLI](#ztp-cli)
+  * [Action Items](#action-items)
+  * [Phases](#phases)
+
+# Revision
+| Rev | Date     |  Author       | Change Description                |
+|:---:|:------------:|:------------------:|-----------------------------------|
+| v0.1 |   11/03/18   |    Simone Salman   | Initial version                   |
+| v0.2 |   11/14/18   |    Simone Salman   | Addressing Review comments        |
+
+# Requirements
+- Build option to have ZTP enabled
+- SNMP should be enabled during the ZTP process (achieved through leveraging updategraph)
+- ZTP should verify, download and install SONiC image
+- Updategraph handles all configuration during ZTP service
+- ZTP should receive post config validation script, that collects switch info and sends it back to a remote location for processing
+- ZTP status should be logged through syslog
+- Interruption of ZTP service should set ZTP status as incomplete/interrupted and ZTP should be disabled.
+
+# DHCP Options
+The following are the private DHCP options used during the implementation of updategraph/ZTP
+
+| DHCP Option | Information         | Explanation                                       |
+|:-----------:|:-------------------|:-------------------------------------------------|
+|    224      | snmp_community     | usage implemented through updategraph |
+|    225      | minigraph_url     | usage implemented through updategraph |
+|    226      | acl_url           | usage implemented through updategraph |
+|    227      | ztp_image_url     | URL for the SONiC image for the switch to dowload |
+|    228      | validation_url    | URL for the validation script -- a script that runs post config and  collects device info and sends it for processing to a remote location designated within the validation script |
```

Comment:
I think we should change the name of this to "post_install_url" - there is no need to limit this to validation - it could do anything that the user chooses.